docs/decisions/arch-diagrams/mobile-create-order.mmd

```diff
@@ -0,0 +1,73 @@
+sequenceDiagram
+    participant User
+    participant MobileApp
+    participant CTBackend as CT Backend
+    participant IAPValidator as IAP Validator (Android/iOS)
+    participant MobileStore as Mobile store (Android/iOS)
+    participant CoCo as Commercetools (CoCo)
+    participant EventSystem
+
+    User->>MobileApp: Starts in-app purchase
+    MobileApp->>CTBackend: Sends purchase info (price, course, platform, token)
+
+    CTBackend->>CoCo: Check if user already has a cart
+    CoCo-->>CTBackend: Return existing cart or none
+    alt Cart exists
+        CTBackend->>CoCo: Delete old cart
+    end
+
+    CTBackend->>CoCo: Add course and external price (in localized currency) to a new cart
+    CoCo-->>CTBackend: Return new cart
+
+    CTBackend->>IAPValidator: Validate the purchase token
```

Comment:
@attiyaIshaque Why have we added validator as a separate service? Can this and `MobileStore` not be considered as just one system, i.e., `MobileStore`?

Reply:
We keep IAPValidator separate because it handles internal logic like retries, redundant payment checks, and platform-specific validation, not just calling the mobile store APIs. This makes the code cleaner, testable, and easier to extend.